packages/manager/apps/web/client/app/domain-operation/domain-operation.service.js

```diff
@@ -66,32 +87,65 @@ angular.module('services').service(
     }
 
     /**
-     * Relaunch an operation
+     * Relaunch a domain operation
      * @param {string} id
      */
-    relaunchOperation(id) {
+    relaunchDomainOperation(id) {
       return this.OvhHttp.post(`/me/task/domain/${id}/relaunch`, {
         rootPath: 'apiv6',
         broadcast: 'domains.operations.relaunched',
       });
     }
 
+    /**
+     * Relaunch a dns operation
+     * @param {string} id
+     */
+    relaunchDnsOperation(id) {
+      return this.OvhHttp.post(`/me/task/dns/${id}/relaunch`, {
+        rootPath: 'apiv6',
+        broadcast: 'domains.operations.relaunched',
+      });
```

Comment:
```suggestion
      return this.$http.post(`/me/task/dns/${id}/relaunch`).then(({data}) => data);
```
And in you controller you can reload your $state.